### DISCUSSION TITLE
Vegetation file always have same coding for the colors

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#8] [Ideas]
Today the vegetation files in one batch execution can vary with what number in DN the different colors have. Would it be possible to define that e.g.

White is always 0
Yellow is always 1
Light green always 2
middle green always 3
dark green always 4

etc.

I understand this has to do with how many colors you have. but ideally if you have the same number of colors, they should all have the same indexed color in the image file.

rational - when using pologynizer and other vectoring tools, you can then batch all vegetation files instead of merging them to one large - and then do a pologynizer method on the large image - which sometimes fails.

**Comment by rphlo** [comment]
Hi Stefan,
I was thinking to look at this now.
Will come back at you when this is ready.

**Comment by stefankinell** [comment]
Cool!
I was about to look into these processes this winter as well to understand if it is possible to export as vector layers directly in the script. Not only as imges. It it is not super important sincece I have another process to solve it in qgis. But it would be handy to get it directly from the script. Both for green/yellow but also from undergrowth. Which is harder to do raster to polygon process on.

So if you see a solution or way to do that in the script let me know. Perhaps it is time for me to lern some rust programing as well. :-)

**Comment by rphlo** [comment]
Kartta-pullautin is very pixel oriented, so outputting vectorised images might be out of the scope.

**Reply by stefankinell** [reply]
That's fine, and good information. Then I probably know that I should not spend to much time into trying to understand the processes and see if it can be adopted/changed. Perhaps to capture undervegetation in a better way I should see if it is easy to make that output being a solid color instead of lines. Then vetorizing can be done in another proccess.

**Comment by rphlo** [comment]
See latest release, I will do something similar for undergrowth later.

**Reply by rphlo** [reply]
And now finalised the feature behind a feature flag

Release:
https://github.com/rphlo/rusty-pullauta/releases/tag/20231214b

**Reply by rphlo** [reply]
Feedbacks welcome
